Sources/Prometheus/PrometheusCollectorRegistry.swift

```diff
@@ -71,16 +71,18 @@ public final class PrometheusCollectorRegistry: Sendable {
     /// created ``Counter`` will be part of the export.
     ///
     /// - Parameter name: A name to identify ``Counter``'s value.
+    /// - Parameter help: Help text for the metric. If a non-empty string is provided, it will be emitted as a `# HELP` line in the exposition format. If the parameter is omitted or an empty string is passed, the `# HELP` line will not be generated for this metric.
     /// - Returns: A ``Counter`` that is registered with this ``PrometheusCollectorRegistry``
-    public func makeCounter(name: String) -> Counter {
+    public func makeCounter(name: String, help: String = "") -> Counter {
```

Comment:
Curious what you think would be the best approach to support the `HELP` line throughout without breaking things. Thought that a default value could solve it.

Reply:
So _technically_ this isn't _100%_ source compatible because folks can get a function handle to a method and that then changes the signature of it which could break folks.

In practice, even Swift is breaking such APIs so given how not absolutely everywhere this API is used I think this is good enough IMHO -- it will be source compatible for the normal usage of calling methods. 

This package does not need to worry about ABI (which this does break), so that's not a problem.

Reply:
Thanks for adding some context @ktoso -- super helpful! Should you come up with a better idea later, happy to quickly change it, just thought the HELP line option should also be available when not using the newer `MetricNameDescriptor`.